include/tvm/ir/type.h

```diff
@@ -164,10 +164,17 @@ class PointerTypeNode : public TypeNode {
   }
 
   bool SEqualReduce(const PointerTypeNode* other, SEqualReducer equal) const {
-    return equal(element_type, other->element_type);
+    // Make "global" equal to ""
+    String lhs_scope = storage_scope.empty() ? "global" : storage_scope;
```

Comment:
Rather than implicitly making `"" == "global"` could we instead change the default `storage_scope` to `"global"` ? WDYT @junrushao1994 ?

Reply:
I agree that should be the cause of action in the long run. There can be some TE legacy setups that we need to fix before making this change. So one thing we can do is to first bring this in, then do an additional refactor later.

Reply:
In principle we should make the default “global” instead of “”

Reply:
I tried the initial commit here with a find/replace of `storage_scope = ""` -> `storage_scope = "global"` and `storage_scope=""` -> `storage_scope="global"` as well as changing `scope` in `decl_buffer` in `python/tvm/tir/buffer.py`. These changes pass tests in `./tests/scripts/task_python_unittest.sh` and `./tests/scripts/task_python_integration.sh`, are there other parts of TE to update @tqchen ?

Reply:
perhaps @Hzfengsy can confirm

Reply:
Thanks @Mousius. Changing the default storage_scope to "global" is exactly what we want in the long run. But there are some TE mechanism based on an empty scope.

I have checked the codes and found that this function will use Stage scope to create `RealizeNode` and further to create a buffer with the stage scope. https://github.com/apache/tvm/blob/41879b2552364f094492470a77a3ec0866b30eae/src/te/schedule/schedule_ops.cc#L43-L56

Reply:
Hi @Hzfengsy, unfortunately I'm not sure I understand the issue here and why it's not caught by the automated tests after making the changes I mentioned, could you provide a test case which illustrates the issue this causes?

Reply:
Sorry for not explaining clearly. To be specific, I choose one test case `tests/python/unittest/test_runtime_module_based_interface.py` for detail explanation:
1. I repeated your changes that you mentioned before. The testcase passed.
2. However, I add `ICHECK(!s->scope.empty())` at `tvm/src/te/schedule/schedule_ops.cc` line 54. It fails.
3. I enforce it to be "global" by changing lines, it also fails:
    ```cpp
    String storage_scope = s->scope.empty() ? "global" : s->scope;
    return s->op->BuildRealize(s, dom_map, pipeline, storage_scope);
    ```

Reply:
Thanks @Hzfengsy, this makes it a lot clearer! This is likely due to some mismatch of the default `scope` in `Stage` - I did a quick experiment in setting the default `scope` to `"global"` in the `Stage` constructor (https://github.com/apache/tvm/blob/main/src/te/schedule/schedule_lang.cc#L92) and this revealed more places within the TE tests which fail. 

I'd propose we have two options:
1. Change the defaults to `"global"` as far as we can here and raise an issue for updating TE  / `Stage` with the recreation steps you've illustrated - given it overwrites the `scope` in the TE `Stage` to `""` it seems reasonably self-contained.
2. Commit the workaround that's been proposed already and raise an issue for updating everything to default to `"global"` in one move.

@Hzfengsy / @tqchen what do you think of these options?

Reply:
I totally agree to raise an issue for updating everything to "global". We can do this refactoring at anytime if we have time. For now, I'd like to commit this workaround, since there is another place to make `""` equals to `"global"`: https://github.com/apache/tvm/blob/41879b2552364f094492470a77a3ec0866b30eae/src/runtime/thread_storage_scope.h#L121-L127

Reply:
What about we always go with "global" in the new code, i.e. the new TensorIR, and make the "" legacy and deprecate this behavior gradually

Reply:
It's great if we use `"global"` in new codes. But we still need to make `""` equals to `"global`" since there are some legacy codes.

Reply:
Yeah I agree

Reply:
Sounds good to me, thanks for taking the time to walk me through this @Hzfengsy 😸